traffic_ops/traffic_ops_golang/deliveryservice/keys.go

```diff
@@ -135,6 +144,48 @@ func GetSSLKeysByHostName(w http.ResponseWriter, r *http.Request) {
 	}
 
 	hostName := inf.Params["hostname"]
+	xmlID, ok, err := getXmlIdFromHostname(inf, hostName)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	if !ok {
+		api.WriteRespAlert(w, r, tc.InfoLevel, "- error getting cdn or delivery service for hostname "+hostName)
+		return
+	}
+
+	getSSLKeysByXMLIDHelper(xmlID, inf, w, r)
+}
+
+// GetSSLKeysByHostName fetches the ssl keys for a deliveryservice specified by the fully qualified hostname. V14 includes expiration date.
+func GetSSLKeysByHostNameV14(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"hostname"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	if inf.Config.RiakEnabled == false {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusServiceUnavailable, errors.New("the Riak service is unavailable"), errors.New("getting SSL keys from Riak by host name: Riak is not configured"))
+		return
+	}
+
+	hostName := inf.Params["hostname"]
+	xmlID, ok, err := getXmlIdFromHostname(inf, hostName)
```

Comment:
So after looking at all this code I think a better way to do the handling below would be to just return non-nil errors from `getXmlIdFromHostname` below instead of logging them at a warning level. Probably by returning "xmlID, userErr, sysErr, errCode". Then you could just make one call to `api.HandleErr` to get all of the logging, proper response and alert levels all at once.

It's up to you, as long as it works properly I don't much care how it's done. Just seems easiest to me.

Reply:
yea i like that better too. done